xray/segment.go

```diff
@@ -286,12 +286,13 @@ func (seg *Segment) Close(err error) {
 		seg.addError(err)
 	}
 
+	seg.Unlock()
+
 	// If segment is dummy we return
 	if seg.Dummy {
 		return
```

Comment:
I would suggest adding seg.Unlock() before this return statement because adding seg.Unlock before this block could cause data race while accessing the seg.Dummy field. Also, would you mind adding the same change in the CloseAndStream method.  (https://github.com/aws/aws-xray-sdk-go/blob/master/xray/segment.go#L319)

Reply:
So basically there will be 2 seg.Unlock call.

Reply:
`CloseAndStream` uses a defer for Unlock so it should be fine. From what I can tell it's not possible to do the same in `Close()` because of the call to `send()`, right?

Reply:
Yeah. Right! Thanks.